src/Crypto/JOSE/Types.hs

```diff
@@ -128,6 +129,14 @@ genSizedBase64IntegerOf :: Int -> Gen SizedBase64Integer
 genSizedBase64IntegerOf n =
   SizedBase64Integer n . bsToInteger <$> genByteStringOf n
 
+-- | Create a 'SizedBase64Integer'' from an 'Integer'.  This is slow since it
+-- does an encoding of the 'Integer' to check the size, so it should be used
+-- with care.
+makeSizedBase64Integer :: Integer -> SizedBase64Integer
+makeSizedBase64Integer x =
+  let bytes = review base64url (integerToBS x) in
```

Comment:
Hm, is this correct? - I think it is the number of _unencoded_ octets.

Reply:
Furthermore, should not need to encode to determine the length - the minimum required length is based on the logarithm of the integer value.

Reply:
See `instance Arbitrary SizedBase64Integer` from `Crypto.JOSE.Types`:
```haskell
arbitrary = do                                                          
  x <- arbitraryBigInteger                                              
  l <- Test.QuickCheck.elements [0,1,2]  -- number of leading zero-bytes
  pure $ SizedBase64Integer ((log2 x `div` 8) + 1 + l) x                
```

Reply:
You're right, sorry, I completely misread that part of the code.